src/main/java/com/amazon/opendistroforelasticsearch/security/rest/TenantInfoAction.java

```diff
@@ -137,7 +144,35 @@ public void accept(RestChannel channel) throws Exception {
             }
         };
     }
-    
+
+    private boolean isAuthorized() {
+        final User user = (User)threadContext.getTransient(ConfigConstants.OPENDISTRO_SECURITY_USER);
+
+        // check if the user is a kibanauser or super admin
+        boolean validateUserByName = (user != null) && (user.getName().equals(evaluator.kibanaServerUsername()) || adminDns.isAdmin(user));
+
+        if (validateUserByName) {
+            return true;
+        }
+
+        // If user check failed by name and admin, check if the users belong to kibana opendistro role
+        final SecurityDynamicConfiguration<?> rolesMappingConfiguration = load(CType.ROLESMAPPING, true);
+
+        // check if kibanaOpendistroRole is present in RolesMapping and if yes, check if user is a part of this role
+        if (rolesMappingConfiguration != null && user != null) {
+            String kibanaOpendistroRole = evaluator.kibanaOpendistroRole();
```

Comment:
What happens if this is empty/null ? Should we add additional checks?

Reply:
returning false in this case.